src/test/java/nl/altindag/log/LogCaptorShould.java

```diff
@@ -379,10 +383,49 @@ void captureMdcHeadersWhereSlf4jIsUsed() {
         Service service = new ServiceWithSlf4jAndMdcHeaders();
         service.sayHello();
 
-        assertMdcHeaders(logCaptor, "test-slf4j-mdc", "hello-slf4j");
+        assertDiagnosticContext(logCaptor, "test-slf4j-mdc", "hello-slf4j");
     }
 
-    private static void assertMdcHeaders(LogCaptor logCaptor, String mdcKey, String mdcValue) {
+    @Test
+    void detachAppenderWithCloseMethod() {
+        Logger logger = (Logger) LoggerFactory.getLogger(this.getClass());
+        assertThat(fetchAppenders(logger)).isEmpty();
+
+        logCaptor = LogCaptor.forClass(this.getClass());
+        assertListAppender(logger);
+
+        logCaptor.close();
+        assertThat(fetchAppenders(logger)).isEmpty();
+    }
+
+    @Test
+    void detachAppenderWithAutoClosable() {
+        Logger logger = (Logger) LoggerFactory.getLogger(this.getClass());
+        assertThat(fetchAppenders(logger)).isEmpty();
+
+        try (LogCaptor ignored = LogCaptor.forClass(this.getClass())) {
+            assertListAppender(logger);
+        }
+
+        assertThat(fetchAppenders(logger)).isEmpty();
+    }
+
+    private static void assertListAppender(Logger logger) {
+        assertThat(fetchAppenders(logger))
+                .hasSize(1)
+                .first()
+                .isInstanceOf(ListAppender.class)
+                .extracting(Appender::getName)
+                .isEqualTo("log-captor");
+    }
+
+    private static List<Appender<?>> fetchAppenders(Logger logger) {
```

Comment:
I think this could be refactored by using streams of java 8. What do you think of the following snippet:
```
private static List<Appender<ILoggingEvent>> fetchAppenders(Logger logger) {
    return StreamSupport.stream(Spliterators.spliteratorUnknownSize(logger.iteratorForAppenders(), Spliterator.ORDERED), false)
            .collect(Collectors.toList());
}
```

Reply:
Okay)